src/app/components/TaskForm.tsx

```diff
@@ -20,32 +20,34 @@ import { PlusCircle } from "lucide-react";
 const Categories_List = ["choose", "Work", "Personal", "Health", "Other"];
 
 interface TaskFormProps {
-  onSubmit: (task: {
-    title: string;
-    description: string;
-    category: string;
-    date: string;
-  }) => void;
+  userId: string;
+  addTask: (formData: FormData) => Promise<void>;
 }
 
-export default function TaskForm({ onSubmit }: TaskFormProps) {
+export default function TaskForm({ userId, addTask }: TaskFormProps) {
   const [task, setTask] = useState({
     title: "",
     description: "",
     category: "choose",
     date: "",
   });
 
-  const handleSubmit = (e: React.FormEvent) => {
-    e.preventDefault();
-    onSubmit(task);
-    setTask({
-      title: "",
-      description: "",
-      category: "choose",
-      date: "",
-    });
-  };
+  async function onSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+    formData.append("userId", userId);
+    try {
+      await addTask(formData);
+      setTask({
```

Comment:
**Suggestion: can you reload the page or have addTask return all the userTasks from the server?**